modmain.lua

```diff
@@ -272,7 +276,17 @@ if DST and SHOWSTATNUMBERS and GLOBAL.kleifileexists("scripts/widgets/wandaageba
 end
 
 local function FindSeasonTransitions()
-	if DST then return {"autumn", "winter", "spring", "summer"} end
+	if DST then
+		local seasons_trans = {"autumn", "winter", "spring", "summer"}
+		--IsShipwreckedWorld and IsPorkWorld are defined in Island Adventures.
+		if HAS_MOD.ISLAND_ADVENTURES then --This can all be one tertiary, but for claritys sake I made this its own statement
+			return GLOBAL.IsShipwreckedWorld() and {"mild", "wet", "green", "dry"}

```

Comment:
I'm not familiar with the IA code-- just making sure that this will be somewhat backwards-compatible? (For example if these functions are not defined then this will error on `attempt to call nil`)

Reply:
It is indeed backwards-compatible in that it will not crash. The `ia_core` field in modinfo being made use of to detect if Island Adventure is enabled is only being set in the version of the rewrite.
Past versions of Island Adventures will not be detected as they do not have ia_core being set in any way.
Hope that helps!